src/task2/medziai/Spygliuotis.java

```diff
@@ -5,12 +5,12 @@
 public class Spygliuotis implements Medis {
     @Override
     public void turi() {
-        switch (this) {
-            case Kadagys kadagys -> System.out.println("Kadagys turi spyglius.");
-            case Pusis pusis -> System.out.println("Pušis turi spyglius.");
-            case Egle egle -> System.out.println("Eglė turi spyglius.");
-            case null, default -> System.out.println("Spygliuotis turi spyglius.");
-        }
+        System.out.println(this);
```

Comment:
Labai arti tiesos! Bet gali padaryti `this + "turi spyglius"` ir visi spygliuočiai turės spyglius:)